### ISSUE TITLE
docs: Experience with initial setup and troubleshooting could be improved

### ISSUE THREAD

**Issue by thelightning87** [docker-mailserver/docker-mailserver#3657]
### 📝 Preliminary Checks

- [X] I tried searching for an existing issue and followed the [debugging docs](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/) advice, but still need assistance.

### 👀 What Happened?

I tried to setup DMS on a clean Debian 12.2 install today but it appears fully broken.

No clients can connect to mailserver on IMAPS/993 and/or SMTPS/587. All get "connection refused"

```console
root@mail:~# netstat -ntlp
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:993             0.0.0.0:*               LISTEN      3347/docker-proxy
tcp        0      0 0.0.0.0:587             0.0.0.0:*               LISTEN      3370/docker-proxy
tcp        0      0 0.0.0.0:25              0.0.0.0:*               LISTEN      3413/docker-proxy
tcp        0      0 0.0.0.0:143             0.0.0.0:*               LISTEN      3392/docker-proxy
tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      8736/docker-proxy
tcp        0      0 0.0.0.0:3306            0.0.0.0:*               LISTEN      5465/docker-proxy
```

- `docker logs <container>` shows no connection attempts.
- No network filtering devices in between client and server. Only routed from subnet x->y
- Connections to arrive on DMS on ports 143 and 25 but still Outlook reports errors about unsupported authentication methods.

Please fix the `:latest`

### 🐋 DMS Version

v12.1.0 and edge

### 💻 Operating System and Architecture

Debian 12.2

### ⚙️ Container configuration files

```bash
docker run -d \
  --name=mailserver \
  --hostname="$HOSTNAME" \
  --domainname="domain.com" \
  -p 25:25 -p 143:143 -p 587:587 -p 993:993 \
  -e ENABLE_SPAMASSASSIN=1 \
  -e SPAMASSASSIN_SPAM_TO_INBOX=1 \
  -e ENABLE_CLAMAV=1 \
  -e ENABLE_POSTGREY=1 \
  -e ENABLE_FAIL2BAN=0 \
  -e ENABLE_SASLAUTHD=0 \
  -e ONE_DIR=1 \
  -e TZ=Europe/Amsterdam \
  -v ~/docker/mailserver/data/:/var/mail/ \
  -v ~/docker/mailserver/state/:/var/mail-state/ \
  -v ~/docker/mailserver/logs/:/var/log/mail/ \
  -v ~/docker/mailserver/config/:/tmp/docker-mailserver/ \
  --restart=unless-stopped \
  mailserver/docker-mailserver
```

### 📜 Relevant log output

Insecure outlook attempt:

```Text
Nov 26 12:57:50 mail dovecot: imap-login: Login: user=<user@domain.com>, method=PLAIN, rip=192.168.20.50, lip=172.17.0.2, mpid=8584, session=<dUEJ5QwLVfnAqBQy>
Nov 26 12:57:50 mail dovecot: imap(user@domain.com)<8584><dUEJ5QwLVfnAqBQy>: Disconnected: Connection closed (IDLE running for 0.001 + waiting input for 0.001 secs, 2 B in + 10 B out, state=wait-input) in=11 out=436 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
Nov 26 12:57:50 mail postfix/postscreen[8440]: CONNECT from [192.168.20.50]:63830 to [172.17.0.2]:25
Nov 26 12:57:50 mail postfix/postscreen[8440]: PASS OLD [192.168.20.50]:63830
Nov 26 12:57:50 mail postfix/smtpd[8441]: connect from unknown[192.168.20.50]
Nov 26 12:57:50 mail postfix/smtpd[8441]: lost connection after EHLO from unknown[192.168.20.50]
Nov 26 12:57:50 mail postfix/smtpd[8441]: disconnect from unknown[192.168.20.50] ehlo=1 commands=1
```

**Comment by thelightning87** [comment]
Nevermind. I switched to iRedMail and everything worked right away so i'm going with that anyway.

**Comment by thelightning87** [comment]
The documention is an absolute incomplete mess by the way. How is anyone supposed to work with this?

**Comment by casperklein** [comment]
> The problem is with the docker-mailserver container.

No it's not. I am running :edge and everything is alright.

> The documention is an absolute incomplete mess by the way. How is anyone supposed to work with this?

It's open source and community driven. Feel free to enhance the "mess".

> Outlook reports errors about unsupported authentication 

Port 25 does not offer authentication. Use one of the following submission ports for that:

```
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
```

> All get "connection refused"

You may have been blocked by fail2ban, because of too much failed attempts.

Try this settings while debugging your connection problems:

```
POSTSCREEN_ACTION=ignore
ENABLE_POSTGREY=0
ENABLE_FAIL2BAN=0
ENABLE_DNSBL=0
```

When making changes, ensure that you use the [correct commands](https://docker-mailserver.github.io/docker-mailserver/latest/usage/#get-up-and-running) when restarting DMS.

**Comment by georglauterbach** [comment]
1. Please [format your code](https://www.markdownguide.org/basic-syntax/#code-blocks) properly.
2. `:edge` is running fine, `:latest` as well, I can assure you that. I am even willing to bet on this.

    > The documention is an absolute incomplete mess by the way. How is anyone supposed to work with this?

3. You are being rude. We have many maintainers that have put great efforts into writing our documentation, and the overwhelming majority regularly tells us our docs are good. The rest is willing to provide a PR.

    You are a newcomer, and by the looks of it, not well-versed with mail servers (see your comments: "Connections to arrive on mailserver on ports 142 and 25 but still Outlook reports errors about unsupported authentication methods." when ports 25 never offered authentication). I would advise you to be more reserved about making claims that DMS is literally not working.

    The maintainers of this project, including me, worked with mail servers for a long time now. But still **you think we ship a stable release to thousands of individuals that is not working?** That is **egregious**!

    I am classifying this as a [violation of the standards set in our code of conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md#our-standards) because I can reasonably consider this report and the following comments as inappropriate in a professional setting. You will comply with our Code of Conduct or I will take action.

6. What @casperklein is the most likely scenario. Unban your IPs and disable F2B to begin with. Here is the documentation link: <https://docker-mailserver.github.io/docker-mailserver/latest/faq/#connection-refused-or-no-response-at-all>. If you had taken the time to **literally search for "connection refused" in the search bar of our documentation**, you would have found help in **the first article that pops up**.

    > fyi; Running iRedMail on the same machine, everything works instantly. The problem is with the docker-mailserver container.

7. iRedMail is super nice and I myself have had the pleasure of working with it. The fact that it works while DMS does not **for you** _does not_ mean DMS is broken in general. If this was the case, we would have literally had **thousands of users** starting to complain in this tracker. As v12.1.0 is out for six month now, and no one (except for you) has had complaints yet, it is very certain that the image is fine - your configuration is the issue.

**Comment by polarathene** [comment]
I've taken the time to respond to your concerns, hopefully it's not a waste of time and you appreciate it.

## Messy Docs

> The documention is an absolute incomplete mess by the way.
> How is anyone supposed to work with this?

As someone who has sunk a hefty amount of hours (_probably in the hundreds by now_) into contributing, revising, and reviewing community contributions for the docs... I'm a bit surprised?

They're by no means perfect, quite a fair amount is from the original Github Wiki before I migrated that into the actual project repo so a review process on docs contributions is in place.
- Pages like the IPv6 docs in v13 have received complete rewrites from my efforts, not for my benefit but our users. I've done similar improvements elsewhere in the docs, usually more iterative.
- I'd love for the docs to be improved but we rarely see much contributions tidying it up and my time is a limited resource that benefits our users and maintainers by addressing other concerns as a priority. When I can afford to, I do iterate on the docs.

My time is completely voluntary on the project, but if you'd like to sponsor some work on improving the docs I'd be happy to arrange that, or review any contributions you can spare time for ❤️ 

Your attitude implies you probably won't do either though 🤷‍♂️ At the very least please dish out criticism in a constructive manner:
- What was lacking with our docs for it to be a mess for you?
- What would you suggest for improving the organization / navigation of the docs (_I'm aware this needs improvement_)
- Were any of the pages lacking in information / clarity for you? How could we better communicate the information you need?
- Where / what were your main frustrations with the docs?
- Were any of the docs outdated or misleading?
  - If your expectations following them were not met, is it possible you jumped ahead and skipped the earlier pages that walk you through proper setup?
  - Did you follow the advice from our debugging docs? (_according to the checkbox above, you have?_) Nothing to add there to help other users in your situation?

FWIW, your report was quite a mess too. No worries, I've edited your report to use proper markdown formatting, so it's much easier for us to parse (_the `docker` CLI command was especially noisy without splitting it into multi-line_).

Please use the proper formatting in any follow-up responses (_assuming you come back, I get the impression you might not_).

---

## Bug Report - Reading comprehension required

You reported version `latest`, which suggests you're not a fan of following instructions (_might be why you had trouble with the docs?_):

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/5c75c16d-6232-4c4d-a659-411fa5b02ba0)

I've changed that to `v12.1.0` for you, since we released `v13.0.0` not long after you raised the bug report. There's a reason we request you avoid `latest` as the version 😅 

---

## Port 25 connection woes

> Connections to arrive on mailserver on ports 142 and 25 but still Outlook reports errors about unsupported authentication methods.

> Insecure outlook attempt:
> ```
> Nov 26 12:57:50 mail dovecot: imap-login: Login: user=<user@domain.com>, method=PLAIN, rip=192.168.20.50, lip=172.17.0.2, mpid=8584, session=<dUEJ5QwLVfnAqBQy>
> ...
> Nov 26 12:57:50 mail postfix/postscreen[8440]: CONNECT from [192.168.20.50]:63830 to [172.17.0.2]:25
> ```

- I've edited your statement to the correct port 143, likely a typo on your end.
- Port 25 is for mail inbound to DMS, or when being sent outbound from DMS. It is intended for MTAs and not mail clients. - Perhaps [our messy docs would help explain this](https://docker-mailserver.github.io/docker-mailserver/v13.0/config/security/understanding-the-ports/) for you if you have time to digest them:
   
   ![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/f2ee7889-a5f5-4432-9855-331df3e7840c)

If you want to login to DMS and send mail you must authenticate with credentials of an account you've configured for DMS.

In your follow-up comment example that is `postmaster@domain.com` with `password`, some clients may expect the user to only be `postmaster` and ignore the domain-part of the mail account.

## Port 587 - Connection refused

As suggested, you're probably running into security measures getting in the way. Why were you enabling extra features like Fail2Ban before you had a working setup? That only adds to your frustration 🤔 

When you're starting out with deploying a new service like DMS, while you're experimenting to get everything working right a good troubleshooting step is to not maintain state. Go to the [DMS docs debugging page](https://docker-mailserver.github.io/docker-mailserver/v13.0/config/debugging/) (that you definitely read):

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/cd4baec3-1ba0-43a5-8c08-62fd2963110a)

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/291ef9be-5c04-4588-bad2-23c9f9ca098a)

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/449e8f2a-9738-4460-866e-4bc07eac5dd8)

You provided a `docker` CLI command as your reproduction and later a `compose.yaml`.
- It doesn't look like it was very minimal, nor did you clarify if you actually tried it again when you encountered such failures.
- It probably wouldn't matter however since you didn't use `--rm` with the CLI command, and you were mounting volumes persisting state.
- Perhaps we could better communicate this in our docs? What would be the least messy way to go about that? Where would you look when things go wrong?

Fun-fact, the debugging docs as you can see above specifically highlight a concern with (re)starting DMS correctly (_although in your case you should disable the volume mounts, at least for `/var/mail-state`_):
- Without `--rm` you'd either explicitly remove the container you named, or start it with `docker start` instead of `docker run`
- With the `compose.yaml`, you may not have been using `docker compose down`, which'd encounter the issues highlighted in the image above.
- Oddly you used `docker exec` with the container ID instead of the name assigned, probably due to inexperience as that sounds inconvenient 🤷‍♂️ I edited that to the more sane approach.

---

## iRedMail

> Running iRedMail on the same machine, everything works instantly.
> The problem is with the `docker-mailserver` container.

Technically the problem is with you (misconfiguration), but we'd love to streamline a happy path in our docs which your experience suggests still has rough edges. If you do reply, let us know where you struggled and encountered friction. Only then can we improve the experience for future users.

Were you new to both?

- Did you install iRedMail via a container, or by [following this Debian guide](https://docs.iredmail.org/install.iredmail.on.debian.ubuntu.html)?
- Do you find [this index page](https://docs.iredmail.org/index.html) useful for docs instead of our tabs + sidebar?
- Here's their [Docker guide](https://github.com/iredmail/dockerized).

That guide shows an interactive guided installation process. I'd love to put something together for DMS, but I can't afford to voluntarily any time soon :(

I'm assuming that your comparing our docs as a mess to the single page experience I linked? Did you have trouble finding [this page](https://docker-mailserver.github.io/docker-mailserver/v13.0/usage/) which is effectively the same process but for DMS? The [README](https://github.com/docker-mailserver/docker-mailserver#link-links-to-useful-resources) and docs try to make it easy to discover:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/89555cc8-9372-4643-9145-2afea5ea25d8)

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/d5d5dad8-030d-4c19-aa6e-bddde898c8b4)

---

Are services like Fail2Ban enabled by default with iRedMail?

Where in the docs do you find this information and if it's possible to disable them? **EDIT:** [Looks like it's GUI oriented](https://docs.iredmail.org/iredmail-easy.getting.start.html#hardware-requirements)? (_and 4GB min memory required_ 🤯 )
